components/aws/sagemaker/tests/integration_tests/component_tests/test_groundtruth_component.py

```diff
@@ -0,0 +1,68 @@
+import pytest
+import os
+import json
+import utils
+from utils import kfp_client_utils
+from utils import sagemaker_utils
+from test_workteam_component import create_workteamjob
+import time
+
+
+@pytest.mark.parametrize(
+    "test_file_dir",
+    [
+        pytest.param(
+            "resources/config/image-classification-groundtruth", marks=pytest.mark.canary_test
+        )
+    ],
+)
+def test_groundtruth_labeling_job(
+    kfp_client, experiment_id, region, sagemaker_client, test_file_dir
+):
+
+    download_dir = utils.mkdir(os.path.join(test_file_dir + "/generated"))
+    test_params = utils.load_params(
+        utils.replace_placeholders(
+            os.path.join(test_file_dir, "config.yaml"),
+            os.path.join(download_dir, "config.yaml"),
+        )
+    )
+
+    # First create a workteam using a separate pipeline and get the name, arn of the workteam created.
+    workteam_name = create_workteamjob(kfp_client, experiment_id, region, sagemaker_client, "resources/config/create-workteam")
```

Comment:
if we are anyway creating a pipeline, why not make it part of this pipeline itself?

Reply:
Why would generated resources ever override each other? They are uniquely named and the components should be capable of running many instances in parallel, or am I missing your point here ?

Ideally, Groundtruth should also be tested with Mturk and public workforce in addition to a private workteam. I don't know if we will support those cases in the future but from that point of view this modularity made more sense to me. However, I too considered this design a bit, and am open to changing it.

Reply:
> Why would generated resources ever override each other?

each test's config file resides in its own directory because it generates resources like a config file with placeholders replaced and same directory is used to download outputs from minio, so `resources/config/create-workteam/generated directory` will be used by both `test_groundtruth_labeling_job` and  `test_workteamjob` at the same time when tests are run in parallel

Reply:
Changed the download dir to be a param to workteam creation which should fix this issue.